src/functor.jl

```diff
@@ -169,7 +169,52 @@ function _default_walk(f, x)
   func, re = functor(x)
   re(map(f, func))
 end
-_default_walk(f, ::Nothing, ::Nothing) = nothing
+_default_walk(_, ::Nothing, ::Nothing) = nothing
+
+# Side effects only, saves a restructure
+function _foreach_walk(f, x)
+  foreach(f, children(x))
+  return x
+end
+
+### WARNING: the following is unstable internal functionality. Use at your own risk!
```

Comment:
I considered `_`-prefixing everything but didn't for now because these will probably become part of the public API at some point. Happy to prefix them if that's desired though.